cognee/tasks/schema/ingest_database_schema.py

```
@@ -0,0 +1,104 @@
+        db_provider=database_config.get("migration_db_provider", "sqlite"),
+    )
+    schema = await engine.extract_schema()
+    tables = {}
+    sample_data = {}
+    schema_tables = []
+    schema_relationships = []
+
+    async with engine.engine.begin() as cursor:
+        for table_name, details in schema.items():
+            rows_result = await cursor.execute(
+                text(f"SELECT * FROM {table_name} LIMIT {max_sample_rows}")
```

Ruff (0.12.2)

Possible SQL injection vector through string-based query construction

(S608)